src/lib/arrays.js

```diff
@@ -0,0 +1,112 @@
+const getNthElement = (index, array) => {
+  if (index >= array.length) {
+    const newIndex = index - array.length;
+    return array[newIndex];
+  }
+  return array[index];
+};
+
+const arrayToCSVString = array => {
+  return array.toString();
+};
+
+const csvStringToArray = string => {
+  return string.split(',');
+};
+
+const addToArray = (element, array) => {
+  array.push(element);
+};
+
+const addToArray2 = (element, array) => {
+  const newArray = [...array];
+  newArray.push(element);
+  return newArray;
+};
+
+const removeNthElement = (index, array) => {
+  return array.splice(index, 1);
+};
+
+const numbersToStrings = numbers => {
+  const newString = numbers.toString();
+  return newString.split(',');
+};
+
+const uppercaseWordsInArray = strings => {
+  // return strings.toString().toUpperCase().split(",");
+  return strings.map(string => {
+    return string.toUpperCase();
+  });
+};
+
+const reverseWordsInArray = strings => {
+  return strings.map(word => {
+    return word
+      .split('')
+      .reverse()
+      .join('');
+  });
+};
+
+const onlyEven = numbers => {
+  return numbers.filter(number => {
+    return number % 2 === 0;
+  });
+};
+
+const removeNthElement2 = (index, array) => {
+  return array.filter((element, i) => {
+    return i !== index;
+  });
+};
+
+const elementsStartingWithAVowel = strings => {
+  return strings.filter(string => {
+    const vowels = 'aeiou';
+    const lowerCaseString = string.toLowerCase();
+    const firstLetter = lowerCaseString[0];
+    return vowels.includes(firstLetter);
+  });
+};
+
+const removeSpaces = string => {
+  return string.split(' ').join('');
+};
+
+const sumNumbers = numbers => {
+  return numbers.reduce((acc, number) => {
+    return acc + number;
+  });
+};
+
+const sortByLastLetter = strings => {
+  const alphabet = 'abcdefghijklmnopqrstuvwxyz';
+  return strings.sort((current, next) => {
+    const currentLastLetterIndex = current.length - 1;
+    const currentLastLetter = current[currentLastLetterIndex];
+    const nextLastLetterIndex = next.length - 1;
+    const nextLastLetter = next[nextLastLetterIndex];
+    const currentIndexInAlphabet = alphabet.indexOf(currentLastLetter);
+    const nextIndexInAlphabet = alphabet.indexOf(nextLastLetter);
+    return currentIndexInAlphabet - nextIndexInAlphabet;
```

Comment:
This is an interesting way to go about this! Theres many different ways you could solve this, so rather than go into them all here Ill just stick to what you could do to refactor this version.
There are a lot of variables created here that are only used once, which is generally bad practice. Looking through your function and putting the variables values in where they are used, it can be boiled down to just this:
```
    return alphabet.indexOf(current[current.length - 1]) - alphabet.indexOf(next[next.length - 1]);
```
Interesting approach as I say, good to see you working these things out for yourself and hitting your won conclusions!